### PR TITLE
Fix tests and apply black

### PR DESCRIPTION
**Pull request by rmoyard**
1. Change the error message raised in one test and remove the previously raised error as now it is done in PennyLane.
2. Remove the tests of gates that were removed.
3. Apply black

Non black changes are marked.

### CODE REVIEW

File: tests/test_apply.py [hunk 1]
```diff
@@ -216,7 +227,7 @@ def test_invalid_qubit_state_vector(self, device):
         state = np.array([0, 123.432])
         wires = [0, 1]
 
-        with pytest.raises(ValueError, match=r"State vector must be of length 2\*\*wires"):
+        with pytest.raises(ValueError, match=r"State vector must have shape"):
```
Comment (rmoyard) [inline tests/test_apply.py:230]: Important change.

File: tests/test_converter.py [hunk 2]
```diff
@@ -715,45 +719,6 @@ def test_operation_transformed_into_qubit_unitary(self, recorder):
         assert np.array_equal(recorder.queue[0].parameters[0], ex.CHGate().to_matrix())
         assert recorder.queue[0].wires == Wires([0, 1])
 
-    def test_qiskit_gates_to_be_deprecated(self, recorder):
```
Comment (rmoyard) [inline tests/test_converter.py:718]: Important change.

File: pennylane_qiskit/qiskit_device.py [hunk 3]
```diff
@@ -351,9 +349,6 @@ def qubit_state_vector_check(self, operation, par, wires):
                     "is not supported on the unitary simulator backend."
                 )
 
-            if len(par[0]) != 2 ** len(wires):
```
Comment (rmoyard) [inline pennylane_qiskit/qiskit_device.py:354]: Important change.

**Comment by rmoyard** [comment]
[sc-33451]

**Review by lillian542 — APPROVED** [review]
Looks good to me 🎉

**Pull request merged**